sqlserver-delta-plugins/src/e2e-test/java/io.cdap.plugin/BQValidation.java

```diff
@@ -0,0 +1,232 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+
+  /**
+   * Extracts entire data from source and target tables.
+   *
+   * @param sourceTable table at the source side
+   * @param targetTable table at the sink side
+   * @return true if the values in source and target side are equal
+   */
+
+  public static boolean validateDBToBQRecordValues(String schema, String sourceTable, String targetTable)
+    throws SQLException, ClassNotFoundException, IOException, InterruptedException, ParseException {
+    List<Object> targetBigQueryRows = new ArrayList<>();
+    List<JsonObject> targetBigQueryResponse = new ArrayList<>();
+
+    getBigQueryTableData(targetTable, targetBigQueryRows);
+
+    for (Object rows : targetBigQueryRows) {
+      JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
+      targetBigQueryResponse.add(json);
+    }
+
+    String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+    try (Connection connect = MssqlClient.getMssqlConnection();
+         ResultSet rsSource = executeQuery(connect, getSourceQuery)) {
+      return compareResultSetAndJsonData(rsSource, targetBigQueryResponse);
+    }
+  }
+
+  /**
+   * Executes the given SQL query on the provided database connection and returns the result set.
+   *
+   * @param connect The Connection object representing the active database connection.
+   * @param query   The SQL query to be executed on the database.
+   * @return A ResultSet object containing the data retrieved by the executed query.
+   * @throws SQLException If a database access error occurs or the SQL query is invalid.
+   */
+
+  public static ResultSet executeQuery(Connection connect, String query) throws SQLException {
+    connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+    Statement statement = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+    return statement.executeQuery(query);
+    }
+
+  /**
+   * Retrieves the data from a specified BigQuery table and populates it into the provided list of objects.
+   *
+   * @param table        The name of the BigQuery table to fetch data from.
+   * @param bigQueryRows The list to store the fetched BigQuery data.
+   */
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows)
+    throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON (t) FROM (SELECT * EXCEPT( _row_id, _source_timestamp, _is_deleted, " +
+            "_sequence_num) FROM `" + projectId + "." + dataset + "." + table + "`) AS t ORDER BY ID";
```

Comment:
Why are we using EXCEPT in this query don't we need these fields ?

Reply:
Not needed as these are generated in target with these columns other than source which is its functionality.